common/include/pcl/experimental/executor/trait/common_traits.hpp

```diff
@@ -0,0 +1,143 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#include <pcl/type_traits.h>
+
+#include <type_traits>
+
+namespace pcl {
+namespace executor {
+
+/**
+ * \brief In accordance with the named requirement EqualityComparable which
+ * will be formalized in C++20 using concepts
+ * Requirements: https://en.cppreference.com/w/cpp/named_req/EqualityComparable
+ */
+template <typename T1, typename T2, typename = void>
+struct equality_comparable : std::false_type {};
+
+template <typename T1, typename T2>
+struct equality_comparable<
+    T1,
+    T2,
+    pcl::void_t<decltype(std::declval<T1>() == std::declval<T2>(),
+                         std::declval<T2>() == std::declval<T1>(),
+                         std::declval<T1>() != std::declval<T2>(),
+                         std::declval<T2>() != std::declval<T1>())>> : std::true_type {
+};
+
+template <typename T1, typename T2>
+constexpr bool equality_comparable_v = equality_comparable<T1, T2>::value;
```

Comment:
Is this not a trait generic for all types? it should go to the general traits header.

Reply:
I think even `is_instance_of_any` is generic. So should I just delete this file and transfer all the code to the general traits file?

Reply:
you're right. let's go with that yes.